frontend/public/redux.ts

```diff
@@ -53,41 +56,35 @@ const store = createStore(
   composeEnhancers(applyMiddleware(thunk)),
 );
 
-const addPluginListener = () => {
-  const reducerExtensions = pluginStore.getAllExtensions().filter(isReduxReducer);
-  const getReduxFlagsObject = () => {
-    const gatingFlags = getGatingFlagNames(reducerExtensions);
-    const featureState = store.getState()[featureReducerName];
-    return featureState ? _.pick(featureState.toObject(), gatingFlags) : null;
-  };
-
-  let flagsObject = getReduxFlagsObject();
+const pluginReducers: ReducersMapObject = {};
 
-  store.subscribe(() => {
-    const currentFlagsObject = getReduxFlagsObject();
+subscribeToExtensions<ReduxReducer>(
+  extensionDiffListener((added, removed) => {
+    removed.forEach(({ properties: { scope } }) => {
+      delete pluginReducers[scope];
+    });
 
-    if (JSON.stringify(flagsObject) !== JSON.stringify(currentFlagsObject)) {
-      flagsObject = currentFlagsObject;
+    Promise.allSettled(added.map(resolveExtension)).then((results) => {
+      const [fulfilledValues, rejectedValues] = unwrapPromiseSettledResults(results);
 
-      const pluginReducerExtensions = reducerExtensions.filter((e) =>
-        isExtensionInUse(e, flagsObject),
-      );
+      fulfilledValues.forEach(({ properties: { scope, reducer } }) => {
+        pluginReducers[scope] = reducer;
+      });
 
-      const pluginReducers: ReducersMapObject = pluginReducerExtensions.reduce((map, e) => {
-        map[e.properties.namespace] = e.properties.reducer;
-        return map;
-      }, {});
+      if (rejectedValues.length > 0) {
+        // eslint-disable-next-line no-console
+        console.error('Failed to resolve Redux reducer extensions', rejectedValues);
```

Comment:
This is already handled by `useResolvedExtensions` hook in latest #8224 update.

Reply:
but were not using this hook here, `useResolvedExtensions`